patches/tModLoader/Terraria.ID/ProjectileID.cs.patch

```diff
@@ -0,0 +1,41 @@
+--- src/Terraria\Terraria.ID\ProjectileID.cs
++++ src/tModLoader\Terraria.ID\ProjectileID.cs
+@@ -1,3 +_,4 @@
++using ReLogic.Reflection;
+ using System;
+ 
+ namespace Terraria.ID
+@@ -1199,5 +_,32 @@
+ 		public const short BookStaffShot = 712;
+ 		public const short DD2Win = 713;
+ 		public const short Count = 714;
++
++		public static readonly IdDictionary Search = IdDictionary.Create<ProjectileID, short>();
++
++		public static string GetUniqueId(int projType) {
```

Comment:
I think `GetUniqueId` is ambigious, people tend to associate numbers with Ids. I prefer it being renamed to `GetStringRepresentation`

Reply:
I'm currently using `GetReadableUniqueKey(...)`, based on Chicken Bones suggestion. Feel free to suggest an alternative.